File: Solution/CSharp/118.Pascal'sTriangle.cs
static public List<List<int>> Generate(int numRows) {
    List<List<int>> result = new List<List<int>>();
    for (int line = 1; line <= numRows; line++) {
        List<int> list = new List<int>();
        if (line >= 3) {
            list.Add(1);
            int index = 1;
            while (index < line - 1) {
                list.Add(result[line - 2][index - 1] + result[line - 2][index]);
                index++;
            }
            list.Add(1);
        } else if (line == 2) {
            list.Add(1);
            list.Add(1);
        } else if (line == 1) {
            list.Add(1);
        }
        result.Add(list);
    }
    return result;
}

File: Solution/Python/118.Pascal'sTriangle.py
class Solution:
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res = []
        for i in range(1, numRows + 1):
            if i == 1:
                line = [1]
            elif i == 2:
                line = [1, 1]
            else:
                line = [1]
                idx = 1
                while idx < i - 1:
                    line.append(res[i - 2][idx - 1] + res[i - 2][idx])
                    idx += 1
                line.append(1)
            res.append(line)
        return res


numRows = 5
res = Solution().generate(numRows)
print(res)
